### PR TITLE
[android] [file-system] Support file schemes on Android

### PR DESCRIPTION
**Pull request by hbiede**
# Why

#12101 was only partially resolved, so this PR aims to fix the same problem within `expo-file-system`

# How

Replicated an existing workaround we are using in an expo project

# Test Plan

Attempted to save and reopen a file (namely persisted data for redux). This has been broken on Android for a while, but this fix was able to re-instate this functionality.

# Checklist

- [X] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [X] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.kt [hunk 1]
```diff
@@ -82,6 +82,16 @@ private const val MIN_EVENT_DT_MS: Long = 100
 private const val HEADER_KEY = "headers"
 private const val DIR_PERMISSIONS_REQUEST_CODE = 5394
 
+private fun slashifyFilePath(path: String?): String? {
+  return if (path == null) {
+    null
+  } else if (path.startsWith("file:///")) {
+    path
+  } else {
+    path.replace("file:/", "file:///")
```
Comment (tsapeta) [inline packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.kt:91]: What if it starts with `file://`?

File: packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.kt [hunk 2]
```diff
@@ -82,6 +82,16 @@ private const val MIN_EVENT_DT_MS: Long = 100
 private const val HEADER_KEY = "headers"
 private const val DIR_PERMISSIONS_REQUEST_CODE = 5394
 
+private fun slashifyFilePath(path: String?): String? {
```
Comment (tsapeta) [inline packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.kt:85]: I would call it `normalizeFilePath`, but I'm also ok with slashify

File: docs/pages/versions/v44.0.0/sdk/filesystem.md [hunk 3]
```diff
@@ -796,19 +796,19 @@ Alias to [FileSystem.copyAsync(options)](#filesystemcopyasyncoptions)
...
+| `readDirectoryAsync`      | `file:///`                                                                                                                                 | `file://`                                                                                       |
+| `downloadAsync`           | Source:<br/>`http://`,<br/>`https://`<br/><br/>Destination:<br/>`file:///`                                                                 | Source:<br/>`http://`,<br/>`https://`<br/><br/>Destination:<br/>`file://`                       |
+| `uploadAsync`             | Source:<br/>`file:///`<br/><br/>Destination:<br/>`http://`<br/>`https://`                                                                  | Source:<br/>`file://`<br/><br/>Destination:<br/>`http://`<br/>`https://`                        |
+| `createDownloadResumable` | Source:<br/>`http://`,<br/>`https://`<br/><br/>Destination:<br/>`file:///`                                                                 | Source:<br/>`http://`,<br/>`https://`<br/><br/>Destination:<br/>`file://`                       |     |
```
Comment (tsapeta) [inline docs/pages/versions/v44.0.0/sdk/filesystem.md:811]: Could you update https://github.com/expo/expo/blob/main/docs/pages/versions/unversioned/sdk/filesystem.md instead? The `unversioned` is used to generate the docs for future releases 😉 We won't release this change to existing SDK44, so we also shouldn't update its docs.

**Review by tsapeta — CHANGES_REQUESTED** [review]

File: packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.kt (hunk 2 above)
Comment (hbiede) [reply packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.kt:85]: For this and the last, I was basically just stealing the function from [this](https://github.com/expo/expo/pull/12439/files#diff-764e029567150bffdda3518243fc2b67b050121e9effef1bfe8579e532637ffe) 😅

**Review by tsapeta — APPROVED** [review]

**Comment by hbiede** [comment]
No idea why the iOS suite is failing...

**Review by bbarthec — APPROVED** [review]
Looking good to me 👍 
Thank you for taking care of this 🤩

**Comment by bbarthec** [comment]
@hbiede, don't worry about this CI failure, that's unrelated to your changes 👀

**Pull request merged**